### PR TITLE
removed face from halfedge

### PR DESCRIPTION
**Pull request by pca006132**
Just some cleanup, I think we talked about `face` being redundant because we can just compute it from the index, so I took the time to remove it.

### CODE REVIEW

File: src/manifold/src/impl.cpp [hunk 1]
```diff
@@ -496,8 +498,9 @@ void Manifold::Impl::CreateHalfedges(const Vec<ivec3>& triVerts) {
       if (h0.startVert != h1.endVert || h0.endVert != h1.startVert) break;
       if (halfedge_[NextHalfedge(pair0)].endVert ==
           halfedge_[NextHalfedge(pair1)].endVert) {
-        h0.face = -1;
-        h1.face = -1;
+        // FIXME: this actually does nothing
```
Comment (pca006132) [inline src/manifold/src/impl.cpp:501]: This is an interesting one, I was trying to use a vector instead for this and was having weird issues. It turns out we declared `h0` and `h1` as `Halfedge`, so these assignments aren't doing anything. Changing them to `Halfedge&` breaks things, so I just commented them out for now.

File: src/manifold/src/quickhull.cpp [hunk 2]
```diff
@@ -78,13 +78,15 @@ size_t MeshBuilder::addHalfedge() {
     return index;
   }
   halfedges.push_back({});
+  halfedgeToFace.push_back(0);
```
Comment (pca006132) [inline src/manifold/src/quickhull.cpp:81]: Face index is needed in `quickhull`, so I keep a `halfedgeToFace` vector for that.

File: src/manifold/src/sort.cpp [hunk 3]
```diff
@@ -256,21 +231,33 @@ void Manifold::Impl::Finish() {
...
 #ifdef MANIFOLD_DEBUG
+  auto MaxOrMinus = [](int a, int b) {
+    return std::min(a, b) < 0 ? -1 : std::max(a, b);
+  };
+  int face = 0;
   Halfedge extrema = {0, 0, 0, 0};
-  extrema = reduce(halfedge_.begin(), halfedge_.end(), extrema, Extrema());
-#endif
-
+  for (size_t i = 0; i < halfedge_.size(); i++) {
```
Comment (pca006132) [inline src/manifold/src/sort.cpp:239]: Simplified the code a bit as well. Was thinking about using pair iterators, but it is too much work. We don't need parallelization here anyway.

File: src/manifold/src/impl.cpp (hunk 1 above)
Comment (elalish) [reply src/manifold/src/impl.cpp:501]: Ha, I think I put this in to try to debug something and then forgot about it. Let's remove.

File: src/manifold/src/impl.cpp [hunk 4]
```diff
@@ -513,13 +516,13 @@ void Manifold::Impl::CreateHalfedges(const Vec<ivec3>& triVerts) {
...
-      halfedge_[pair0] = {-1, -1, -1, -1};
-      halfedge_[pair1] = {-1, -1, -1, -1};
-    } else {
-      halfedge_[pair0].pairedHalfedge = pair1;
-      halfedge_[pair1].pairedHalfedge = pair0;
-    }
+    // if (halfedge_[pair0].face < 0) {
+    //   halfedge_[pair0] = {-1, -1, -1, -1};
+    //   halfedge_[pair1] = {-1, -1, -1, -1};
+    // } else {
```
Comment (elalish) [inline src/manifold/src/impl.cpp:522]: Same as above.

File: src/manifold/src/quickhull.cpp [hunk 5]
```diff
@@ -295,9 +315,9 @@ std::pair<Vec<Halfedge>, Vec<vec3>> QuickHull::buildMesh(double epsilon) {
              he.endVert = counts[he.endVert];
            });
   // setting face id
-  for (size_t index = 0; index < halfedges.size(); index++) {
-    halfedges[index].face = index / 3;
-  }
+  // for (size_t index = 0; index < halfedges.size(); index++) {
+  //   halfedges[index].face = index / 3;
```
Comment (elalish) [inline src/manifold/src/quickhull.cpp:319]: remove?

**Review by elalish — APPROVED** [review]
Thanks for the cleanup! I'm guessing this didn't make any noticeable speed change?

**Comment by elalish** [comment]
Yeah, I thought about that when I made `Halfedge` - I think it's worth leaving start and end in. It's also useful in other places, like the Boolean where the edges get created and operated on before they know what their pair is.

**Pull request merged**